src/routes/eventstream.rs

```diff
@@ -1,39 +1,48 @@
 use axum::extract::State;
+use axum::http::HeaderMap;
 use axum::response::sse::{Event, Sse};
 use axum::routing::get;
 use axum::Router;
-use chrono::Utc;
-use futures::stream::Stream;
+use futures::stream::{self, Stream};
 use futures::StreamExt;
 use tokio_stream::wrappers::BroadcastStream;
 
 use crate::error::ApiResult;
 use crate::server::appstate::AppState;
 
 pub async fn get_clip_v2(
+    headers: HeaderMap,
     State(state): State<AppState>,
 ) -> Sse<impl Stream<Item = ApiResult<Event>>> {
     let hello = tokio_stream::iter([Ok(Event::default().comment("hi"))]);
+    let last_event_id = headers.get("last-event-id");
 
-    let mut prev_ts = Utc::now().timestamp();
-    let mut idx = 0;
-
-    let channel = state.res.lock().await.hue_channel();
+    let channel = state.res.lock().await.hue_event_stream().subscribe();
+    let stream = BroadcastStream::new(channel);
+    let events = match last_event_id {
+        Some(id) => {
+            let previous_events = state
+                .res
+                .lock()
+                .await
+                .hue_event_stream()
+                .events_sent_after_id(id.to_str().unwrap());
+            stream::iter(previous_events.into_iter().map(Ok))
+                .chain(stream)
+                .boxed()
+        }
+        None => stream.boxed(),
+    };
 
-    let stream = BroadcastStream::new(channel).map(move |e| {
-        let json = [e?];
+    let stream = events.map(move |e| {
+        let evt = e?;
+        let evt_id = evt.id();
```

Comment:
Let's just write `evt.id()` inline below? Or does the borrow checker not grok that?

Reply:
Yep, doesn't work because of the move that happens on the next line.